File: Ex3.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func reverseNumber(num int) int {
	result := 0
	for num != 0 {
		digit := num % 10
		if result != 0 || digit != 0 {
			result = result*10 + digit
		}
		num /= 10
	}
	return result
}

func reverseAndPrintNumbers(numbers []int) {
	for _, num := range numbers {
		fmt.Println(reverseNumber(num))
	}
}

func main() {
	fmt.Println("Введите количество чисел:")
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())
	numbers := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Printf("Введите число #%d:\n", i+1)
		scanner.Scan()
		numbers[i], _ = strconv.Atoi(scanner.Text())
	}
	reverseAndPrintNumbers(numbers)
}

File: Ex3.java
import java.util.Scanner;

public class Main {
    public static int reverseNumber(int num) {
        int result = 0;
        while (num != 0) {
            int digit = num % 10;
            if (result != 0 || digit != 0) {
                result = result * 10 + digit;
            }
            num /= 10;
        }
        return result;
    }

    public static void reverseAndPrintNumbers(int n, int[] numbers) {
        for (int i = 0; i < n; i++) {
            System.out.println(reverseNumber(numbers[i]));
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Введите количество чисел:");
        int n = scanner.nextInt();
        int[] numbers = new int[n];
        for (int i = 0; i < n; i++) {
            System.out.printf("Введите число #%d:\n", i + 1);
            numbers[i] = scanner.nextInt();
        }
        reverseAndPrintNumbers(n, numbers);
    }
}